packages/web-worker/src/test/e2e.test.ts

```diff
@@ -520,6 +520,59 @@ describe('web-worker', () => {
       expect(page.workers()).toHaveLength(0);
     });
   });
+
+  it('throws an error when calling a function on a terminated worker', async () => {
```

Comment:
The other necessary test here is that calling it from the main thread actually releases all relevant memory and that `page.workers()` is empty

Reply:
Added a [test](https://github.com/Shopify/quilt/pull/1111/files#diff-734ac9de6755db022949065ed7bfa5efR609-R686) to verify that `page.workers()` and all the relevant memory is released.